File: tessoku/rust/src/bin/A28.rs
use proconio::input;

fn main() {
    input! {
        n: usize,
        ta: [(char, i64); n],
    }

    let div = 10000;

    let mut res = 0;
    for (t, a) in ta.iter() {
        match t {
            '+' => res += a,
            '-' => res -= a,
            '*' => res *= a,
            _ => (),
        }

        if res < 0 {
            res += div;
        }
        res %= div;

        println!("{}", res);
    }
}

File: tessoku/python/A28.py
n = int(input())

num = 0
for i in range(n):
    t, a = input().split()
    aa = int(a)

    match t:
        case "+":
            num += aa
        case "-":
            num -= aa
        case "*":
            num *= aa

    if num < 0:
        num += 10000

    num %= 10000
    print(num)
